rclpy/rclpy/parameter_client.py

```diff
@@ -0,0 +1,310 @@
+
+            client = AsyncParameterClient(node, 'example_node')
+
+            # set parameters
+            future = client.set_parameters([
+                Parameter('int_param', Parameter.Type.INTEGER, 88).to_parameter_msg(),
+                Parameter('string/param', Parameter.Type.STRING, 'hello world').to_parameter_msg(),
+            ])
+            self.executor.spin_until_future_complete(future)
+            results = future.result()  # rcl_interfaces.srv.SetParameters.Response
+
+        For more on service names, see: `ROS 2 docs`_.
+
+        .. _ROS 2 docs: https://docs.ros.org/en/rolling/Concepts/About-ROS-2-Parameters.html#interacting-with-parameters # noqa E501
+
+        :param node: Node used to create clients that will interact with the remote node
+        :param remote_node_name: Name of remote node for which the parameters will be managed
+        """
+        self.remote_node_name = remote_node_name
+        self.node = node
+        self._get_parameter_client = self.node.create_client(
+            GetParameters, f'{remote_node_name}/get_parameters',
+            qos_profile=qos_profile, callback_group=callback_group
+        )
+        self._list_parameter_client = self.node.create_client(
+            ListParameters, f'{remote_node_name}/list_parameters',
+            qos_profile=qos_profile, callback_group=callback_group
+        )
+        self._set_parameter_client = self.node.create_client(
+            SetParameters, f'{remote_node_name}/set_parameters',
+            qos_profile=qos_profile, callback_group=callback_group
+        )
+        self._get_parameter_types_client = self.node.create_client(
+            GetParameterTypes, f'{remote_node_name}/get_parameter_types',
+            qos_profile=qos_profile, callback_group=callback_group
+        )
+        self._describe_parameters_client = self.node.create_client(
+            DescribeParameters, f'{remote_node_name}/describe_parameters',
+            qos_profile=qos_profile, callback_group=callback_group
+        )
+        self._set_parameters_atomically_client = self.node.create_client(
+            SetParametersAtomically, f'{remote_node_name}/set_parameters_atomically',
+            qos_profile=qos_profile, callback_group=callback_group
+        )
+
+    def service_is_ready(self) -> bool:
+        """
+        Check if all services are ready.
+
+        :return: ``True`` if all services are available, False otherwise.
+        """
+        return all([
+            self._list_parameter_client.service_is_ready(),
+            self._set_parameter_client.service_is_ready(),
+            self._get_parameter_client.service_is_ready(),
+            self._get_parameter_types_client.service_is_ready(),
+            self._describe_parameters_client.service_is_ready(),
+            self._set_parameters_atomically_client.service_is_ready(),
+        ])
+
+    def wait_for_service(self, timeout_sec: Optional[float] = None) -> bool:
```

Comment:
```suggestion
    def wait_for_server(self, timeout_sec: Optional[float] = None) -> bool:
```

or 

```suggestion
    def wait_for_services(self, timeout_sec: Optional[float] = None) -> bool:
```

As long as we're consistent with the `*_ready` method.